compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/expression/FirConventionFunctionCallChecker.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.fir.analysis.checkers.expression
+
+import org.jetbrains.kotlin.KtNodeTypes
+import org.jetbrains.kotlin.fir.FirFakeSourceElementKind
+import org.jetbrains.kotlin.fir.analysis.checkers.context.CheckerContext
+import org.jetbrains.kotlin.fir.analysis.diagnostics.DiagnosticReporter
+import org.jetbrains.kotlin.fir.analysis.diagnostics.FirErrors
+import org.jetbrains.kotlin.fir.analysis.diagnostics.reportOn
+import org.jetbrains.kotlin.fir.analysis.getChild
+import org.jetbrains.kotlin.fir.expressions.FirFunctionCall
+import org.jetbrains.kotlin.fir.references.FirErrorNamedReference
+import org.jetbrains.kotlin.fir.resolve.diagnostics.ConeUnresolvedNameError
+import org.jetbrains.kotlin.util.OperatorNameConventions
+
+object FirConventionFunctionCallChecker : FirFunctionCallChecker() {
+    override fun check(expression: FirFunctionCall, context: CheckerContext, reporter: DiagnosticReporter) {
+        val calleeReference = expression.calleeReference as? FirErrorNamedReference ?: return
+        val diagnostic = calleeReference.diagnostic as? ConeUnresolvedNameError ?: return
+
+        if (expression.calleeReference.source?.kind is FirFakeSourceElementKind) {
+            when (diagnostic.name) {
+                OperatorNameConventions.GET -> reporter.reportOn(expression.source, FirErrors.NO_GET_METHOD, context)
+                OperatorNameConventions.SET -> reporter.reportOn(
+                    expression.source?.getChild(KtNodeTypes.ARRAY_ACCESS_EXPRESSION),
```

Comment:
Mentioned below, but here again. Perhaps this `getChild` is not necessary if we correct the source directly here: https://github.com/JetBrains/kotlin/blob/master/compiler/fir/raw-fir/psi2fir/src/org/jetbrains/kotlin/fir/builder/RawFirBuilder.kt#L1958